WalletWasabi.Tests/UnitTests/BlockchainAnalysis/CoinJoinAnonscoreTests.cs

```diff
@@ -239,4 +240,57 @@ public void InputMergePunishmentNoInheritance()
 		// 10 participants, 1 is you, your anonset would be 10 normally and now too:
 		Assert.Equal(10, tx.WalletOutputs.First().HdPubKey.AnonymitySet);
 	}
+
+	public void SiblingCoinjoinDoesntContributeToAnonScore()
+	{
+		var tx1 = new AnalyzedTransaction();
+		tx1.AddForeignInput();
+		tx1.AddWalletInput();
+		var A = tx1.AddForeignOutput();
+		var B = tx1.AddWalletOutput();
+
+		var tx2 = new AnalyzedTransaction();
+		tx2.AddForeignInput(A);
+		tx2.AddWalletInput(B);
+		tx2.AddForeignOutput();
+		var F = tx2.AddWalletOutput();
+
+		tx1.AnalyzeRecursively();
+
+		Assert.Equal(2, B.Anonymity);
+		Assert.Equal(2, F.Anonymity);
```

Comment:
Concept ACK for this test. Here's a small visualization on what's going on:

![image](https://user-images.githubusercontent.com/9156103/167263751-42bc6948-297f-4691-a0c9-96348aa9b2be.png)